org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLWorkspaceService.java

```diff
@@ -12,36 +12,68 @@
  */
 package org.eclipse.lemminx;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 
+import org.eclipse.lemminx.services.IXMLCommandService;
 import org.eclipse.lsp4j.DidChangeConfigurationParams;
 import org.eclipse.lsp4j.DidChangeWatchedFilesParams;
+import org.eclipse.lsp4j.ExecuteCommandParams;
 import org.eclipse.lsp4j.FileEvent;
 import org.eclipse.lsp4j.SymbolInformation;
 import org.eclipse.lsp4j.WorkspaceSymbolParams;
+import org.eclipse.lsp4j.jsonrpc.CompletableFutures;
+import org.eclipse.lsp4j.jsonrpc.ResponseErrorException;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseError;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseErrorCode;
 import org.eclipse.lsp4j.services.WorkspaceService;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 /**
  * XML workspace service.
  *
  */
-public class XMLWorkspaceService implements WorkspaceService {
+public class XMLWorkspaceService implements WorkspaceService, IXMLCommandService {
+	
+	private static final String WORKSPACE_EXECUTE_COMMAND = "workspace/executeCommand";
 
 	private final XMLLanguageServer xmlLanguageServer;
-
+	
+	private final Map<String, IDelegateCommandHandler> commands;
+	
+	private final Map<String, String> commandRegistrations;
+	
 	public XMLWorkspaceService(XMLLanguageServer xmlLanguageServer) {
 		this.xmlLanguageServer = xmlLanguageServer;
+		this.commands = new HashMap<>();
+		this.commandRegistrations = new HashMap<>();
 	}
 
 	@Override
 	public CompletableFuture<List<? extends SymbolInformation>> symbol(WorkspaceSymbolParams params) {
 		return null;
 	}
-
+	
+	
+	@Override
+	public CompletableFuture<Object> executeCommand(ExecuteCommandParams params) {
+		synchronized (commands) {
+			IDelegateCommandHandler handler = commands.get(params.getCommand());
+			if (handler == null) {
+				throw new ResponseErrorException(new ResponseError(ResponseErrorCode.InternalError, "No command handler for the command: " + params.getCommand(), null));
+			}
+			return CompletableFutures.computeAsync(cancelChecker -> handler.executeCommand(params, cancelChecker));

```

Comment:
IDelegateCommandHandler should throw exception, to manage this exception, I suggest to write:

```java
			return CompletableFutures.computeAsync(cancelChecker -> {
				try {
					return handler.executeCommand(params, cancelChecker);
				} catch (Exception e) {
					if (e instanceof ResponseErrorException) {
						throw (ResponseErrorException) e;
					} else if (e instanceof CancellationException) {
						throw (CancellationException) e;
					}
					throw new ResponseErrorException(
							new ResponseError(ResponseErrorCode.UnknownErrorCode, e.getMessage(), e));
				}
			});
```